app/models/player.rb

```diff
@@ -0,0 +1,31 @@
+class Player < ActiveRecord::Base
+  # has_many :hands
+  has_many :played_cards
+  has_many  :cards, :through => :played_cards
+
+
+  def draw_white_card
+    available_cards = Card.joins("LEFT JOIN played_cards ON played_cards.card_id = cards.id where played_cards.card_id is null and cards.color = 'white'")
```

Comment:
This is a lot of hand written SQL. Can you use an ActiveRecord association and scope for this?

also this line is too long. Try this:

```ruby
available_cards = Card.joins <<-SQL 
	LEFT JOIN 
		played_cards 
	ON 
		played_cards.card_id = cards.id 
	WHERE 
		played_cards.card_id IS null 
	AND 
		cards.color = 'white'
SQL
```